frontend/public/components/graphs/prometheus-poll-hook.ts

```diff
@@ -20,32 +16,7 @@ export const usePrometheusPoll = ({
   timespan = DEFAULT_TIMESPAN,
 }: PrometheusPollProps) => {
   const url = getPrometheusURL({ endpoint, endTime, namespace, query, samples, timeout, timespan });
-  const [error, setError] = useState();
-  const [response, setResponse] = useState();
-  const [loading, setLoading] = useState(true);
-  const safeFetch = useSafeFetch();
-  const tick = useCallback(() => {
-    if (url) {
-      safeFetch(url)
-        .then((data) => {
-          setResponse(data);
-          setError(undefined);
-          setLoading(false);
-        })
-        .catch((err) => {
-          if (err.name !== 'AbortError') {
-            setError(err);
-            setLoading(false);
-            // eslint-disable-next-line no-console
-            console.error(`Error polling Prometheus: ${err}`);
-          }
-        });
-    } else {
-      setLoading(false);
-    }
-  }, [url]);
-
-  usePoll(tick, delay, endTime, query, timespan);
+  const { response, error, loading } = useURLPoll(url, delay, endTime, query, timespan);
```

Comment:
if you use array format and generics for useURLPoll use can do
```suggestion
  return useURLPoll<PrometheusResponse>(url, delay, endTime, query, timespan);
```